net/ampr-ripd/files/ampr-ripd-init

```diff
@@ -0,0 +1,70 @@
+#!/bin/sh /etc/rc.common
+
+START=95
+STOP=10
+
+EXTRA_COMMANDS="configure"
+EXTRA_HELP="        configure       Configure service parameters"
+
+start() {
+   if [ -f /etc/ampr-ripd.conf ]; then
+      if [ -z $(grep -c ^TUNNET=44.127.254/255.255.255.0 /etc/ampr-ripd.conf) ]; then
+         echo "ampr-ripd is not configured
+You must run /etc/init.d/ampr-ripd configure"
+      fi
+   else
+      echo /etc/ampr-ripd.conf not found!  Exiting...
+      exit 1
+   fi
+   if [ ! -d /var/lib/ampr-ripd ]; then
+      mkdir -p /var/lib/ampr-ripd
+   fi
+   source /etc/ampr-ripd.conf
+   ip tunnel change ttl 64 mode ipip tunl0
+   ip link set dev tunl0 up
+   ifconfig tunl0 mtu 1480
+   /usr/sbin/ampr-ripd -s -r -t 44 -i tunl0 -a $TUNNET
+}
+
+stop() {
+   ifconfig tunl0 down
+   killall ampr-ripd
+}
+
+restart() {
+   stop "$@"
+   sleep 1
+   start "$@"
+}
+
+configure() {
+   if [ -z $1 ] | [ -z $2 ] | [ -z $3 ]; then
+      echo Usage: /etc/init.d/ampr-ripd configure [amprhost] [amprmask] [amprnet]
+      exit 1
+   fi
+   amprhost=$1
+   amprmask=$2
+   amprnet=$3
+   echo "Configuring ampr-ripd with values:
+
+amprhost=$amprhost
+amprmask=$amprmask
+amprnet=$amprnet"
+   echo "# Set to your AMPR allocation
+# eg. TUNNET=44.127.254.0/255.255.255.0
+TUNNET=$amprnet/$amprmask" > /etc/ampr-ripd.conf
+   uci set network.amprlan.ipaddr=$amprhost
+   uci set network.amprlan.netmask=$amprmask
+   uci set network.amprwan.ipaddr=$amprhost
+   uci set network.amprwan.netmask=$amprmask
+   for i in $(seq 99 -1 0); do
+     if [ $(uci -q get network.@rule[$i].src) ]; then
+        src=$(uci get network.@rule[$i].src)
+        lookup=$(uci get network.@rule[$i].lookup)
+        if [ "$lookup" = "44" ]; then
+           uci set network.@rule[$i].src=$amprnet/$amprmask
+        fi
+     fi
+   done
+   uci commit network
```

Comment:
Is it enough to set those values and not restart networking?

Reply:
Added message to config process reminding user to restart ampr-ripd and networking.

Reply:
You could also figure out which services are affected and need restarting, or restart the whole networking in the script.

Reply:
For the time being, I'm going to leave it as a manual process, giving the user time to examine settings first.